future-apps/staking/contracts/Staking.sol

```diff
@@ -68,25 +74,26 @@ contract Staking is ERCStaking, AragonApp {
   }
 
   function stakeFor(address acct, uint256 amount, bytes data) authP(STAKE_ROLE, arr(amount)) public {
+    // stake 0 tokens makes no sense
+    require(amount > 0);
     // From needs to be msg.sender to avoid token stealing by front-running
     require(stakingToken.transferFrom(msg.sender, this, amount));
-    processStake(acct, amount, data);
+
+    // process Stake
+    accounts[acct].amount = accounts[acct].amount.add(amount);
+
+    Staked(acct, amount, totalStakedFor(acct), data);
 
     if (stakeScript.length > 0) {
       runScript(stakeScript, data, new address[](0));
     }
   }
 
-  function processStake(address acct, uint256 amount, bytes data) internal {
-    accounts[acct].amount += amount; // overflow check in token contract
-    assert(accounts[acct].amount >= amount);
-
-    Staked(acct, amount, totalStakedFor(acct), data);
-  }
-
   function unstake(uint256 amount, bytes data) authP(UNSTAKE_ROLE, arr(amount)) checkUnlocked(amount) public {
```

Comment:
We should add a function to `unlockAndUnstake` that uses the `unlockAll` function

Reply:
Done.